### PR TITLE
Always sort

### PR DESCRIPTION
**Pull request by greg0ire**

### CODE REVIEW

File: Datagrid/ProxyQuery.php [hunk 1]
```diff
@@ -63,17 +63,38 @@ public function execute(array $params = array(), $hydrationMode = null)
...
+         */
+
+        $identifierFields = $queryBuilder
+            ->getEntityManager()
+            ->getMetadataFactory()
+            ->getMetadataFor(current($queryBuilder->getRootEntities()))
+            ->getIdentifierFieldNames();
+
+        foreach ($identifierFields as $identifierField) {
+            $queryBuilder->addOrderBy(
```
Comment (Soullivaneuh) [inline Datagrid/ProxyQuery.php]: What happen if the querybuilder already have an orderBy?

I think this will be appended, not replaced.

**Comment by Soullivaneuh** [comment]
According to the new [Sonata version management](https://github.com/sonata-project/SonataAdminBundle/issues/3053) and next major [release plan](https://github.com/sonata-project/SonataAdminBundle/issues/3731), this project has been refactored regarding branching and versioning.

If you see this message, your PR concerns a patch or a minor release and is not targeting the right branch.

So I'm closing this one, but don't see it as a refusal. If you think your work is still relevant and want to continue, feel free to reopen it on the right branch (e.g. the default one).

Regards.

File: Datagrid/ProxyQuery.php (hunk 1 above)
Comment (greg0ire) [inline Datagrid/ProxyQuery.php]: Yes, and it is intended.

Comment (greg0ire) [inline Datagrid/ProxyQuery.php]: For instance, imagine you have a sort on the date, but two people happen to post something at the same exact second. You don't want the request to return things in a different order when you press F5, and a good way to achieve that is to append a sort on a unique key

Comment (Soullivaneuh) [inline Datagrid/ProxyQuery.php]: Yes but what about `$this->getSortOrder()`?

Comment (greg0ire) [inline Datagrid/ProxyQuery.php]: As said in the comment "reusing the sort order is the most natural way to go". If you sort by date ASC, and then by id ASC, and you press the button that inverts the sort order, you expect items that have the same date to appear in reverse order. Reusing `getSortOrder` allows that.

**Comment by greg0ire** [comment]
See #544

**Pull request closed without merging**